filters/filtertest/filtertest.go

```diff
@@ -61,6 +64,7 @@ func (fc *Context) BackendUrl() string                  { return fc.FBackendUrl
 func (fc *Context) OutgoingHost() string                { return fc.FOutgoingHost }
 func (fc *Context) SetOutgoingHost(h string)            { fc.FOutgoingHost = h }
 func (fc *Context) Metrics() filters.Metrics            { return fc.FMetrics }
+func (fc *Context) Tracer() opentracing.Tracer          { return &opentracing.NoopTracer{} }
```

Comment:
we can just return the FTracer field, and then the anyone writing tests can decide to set it or not. The other fields work also like this.

Reply:
Anyone needs it in the FilterContext, so I went for returning a valid tracer. We could check that FTracer is non-nil and return that and fall back to the NoopTracer.